compiler/rustc_typeck/src/check/fn_ctxt/checks.rs

```diff
@@ -674,8 +678,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         self.emit_coerce_suggestions(
                             &mut err,
                             &provided_args[input_idx],
-                            final_arg_types[input_idx].map(|ty| ty.0).unwrap(),
-                            final_arg_types[input_idx].map(|ty| ty.1).unwrap(),
+                            provided_ty,
+                            // FIXME(compiler-errors): expected_ty?
+                            final_arg_types[input_idx]
```

Comment:
This should actually probably be `expected_ty`, but this will just provide a bad suggestion, and not panic.

Reply:
What about changing `emit_coerce_suggestions` to take an `Option` here instead (and make it not emit the invalid suggestion when it is `None`)?

Reply:
but emit_coerce_suggestions can't do anything if we don't have a target type to coerce to, right?